### PR TITLE
[release-4.9] Bug 2029861: backport the change from #10394 to release 4.9

### PR DESCRIPTION
**Pull request by yangcao77**
Signed-off-by: Stephanie <yangcao@redhat.com>
This PR ports over change from #10394 to release 4.9 to fix https://bugzilla.redhat.com/show_bug.cgi?id=2025592
The current devfile handler for 4.9 does not work for devfile v2.2
According to the discussion on slack: https://coreos.slack.com/archives/C017YDKHQV6/p1637746963035100

### CODE REVIEW

**Comment by rottencandy** [comment]
/retitle [release-4.9] Bug 2025592: backport the change from #10394 to release 4.9

**Comment by rohitkrai03** [comment]
/retitle [release-4.9] Bug 2029861: backport the change from #10394 to release 4.9

File: vendor/OWNERS [hunk 1]
```diff
@@ -1,8 +0,0 @@
-reviewers:
```
Comment (rottencandy) [inline vendor/OWNERS:1]: I think this file shouldn't be deleted.

**Comment by yangcao77** [comment]
This PR also includes openshift#10464 and openshift#10570 for 4.9. contributed by @gruselhaus

**Comment by yangcao77** [comment]
@spadgett @invincibleJai @rottencandy 
May I get a `approve` label and also a `lgtm` label for this PR?

**Review by rohitkrai03 — APPROVED** [review]
/lgtm

**Comment by sanketpathak** [comment]
Verified pr in cluster created using cluster-bot
Screenshot from 2022-01-21 23-13-18
/label cherry-pick-approved

**Pull request merged**